api/src/main/java/org/pmiops/workbench/db/model/DataSet.java

```diff
@@ -47,12 +51,28 @@ public DataSet(
     this.dataSetId = dataSetId;
     this.workspaceId = workspaceId;
     this.name = name;
+    this.version = DataSet.INITIAL_VERSION;
     this.description = description;
     this.creatorId = creatorId;
     this.creationTime = creationTime;
     this.invalid = invalid;
   }
 
+  public DataSet(DataSet dataSet) {
+    setName(dataSet.getName());
+    setVersion(DataSet.INITIAL_VERSION);
+    setDescription(dataSet.getDescription());
+    setInvalid(dataSet.getInvalid());
+    setIncludesAllParticipants(dataSet.getIncludesAllParticipants());
+    setValues(
+        dataSet.getValues().stream()
+            .map(
+                dataSetValue ->
+                    new DataSetValues(dataSetValue.getDomainId(), dataSetValue.getValue()))
```

Comment:
You could also make a copy constructor for `DataSetValues`, which would simplify the lambda in the `map` call to `DatasetValues::new`.

Reply:
Sure! I'll make a copy constructor.